pkg/controllers/management/aks/aks_cluster_handler_test.go

```diff
@@ -77,7 +68,7 @@ func Test_onClusterChange_Default(t *testing.T) {
 	if err != nil {
 		t.Errorf("error running onClusterChange: %s", err)
 	}
-	if cluster.Status.Conditions[0].Status != mockConditionUnknown.Status {
+	if !IsProvisionedStatusUnknown(cluster) {
```

Comment:
Sorry let me be more explicit, use these specific conditions right there: https://github.com/rancher/rancher/blob/6f86d4a6b0effbaaecadf183ab8888786ca7b6a4/pkg/controllers/provisioningv2/rke2/common.go#L76-L77
Any time you are indexing conditions (like `cluster.Status.Conditions[x]`) is almost certainly not what you should be doing. The `Condition` objects themselves allow you to write `rke2.Provisioned.IsTrue(cluster)` which will do exactly what you want, but the ordering of the conditions no longer matters. The `Condition` object (in this case `Provisioned` just loops all the conditions until it finds itself. Also I'm like 90% positive that those objects are the ones actually used for kev2 hosted provisioning, so it's probably better to test with the condition objects we are using instead of making specific ones for testing. This is also good future proofing in case we have to ever update the test data and add more conditions.